client/src/glossary/GlossaryMenuController.js

```diff
@@ -0,0 +1,156 @@
+import _ from "lodash";
+import React from "react";
+
+import { withRouter } from "react-router-dom";
+
+import { glossaryEntryStore } from "src/models/glossary";
+
+import { SidebarButton } from "src/components";
+
+import { get_glossary_items_by_letter } from "./glossary_utils";
+
+import { GlossaryMenu } from "./GlossaryMenu";
+
+const ROUTES_WITHOUT_GLOSSARY = {
+  "/start": true,
+  "/glossary": true,
+};
+
+const GlossaryMenuController = withRouter(
+  class GlossaryMenuController extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        show_sidebar: true,
+        showGlossary: false,
+        glossaryItem: null,
+        showList: true,
+        results: [],
+        query: "",
```

Comment:
Some of this state (`results` at least, maybe `showList`) can be derived inside child components rather than existing as top-level state here, I believe. Don't think `results` needs to be state _anywhere_, the child components can just derive it from the `query` value (as long as that's memoized, no problem at all).